src/cr/cube/matrix/cubemeasure.py

```diff
@@ -1105,27 +1140,21 @@ def baseline(self):
         return uncond_row_margin / uncond_table_margin
 
     @lazyproperty
-    def columns_margin(self):
-        """1D ndarray of np.float64 weighted N for each matrix column."""
-        # --- only selected counts contribute to the columns margin, which is summed
+    def column_bases(self):
+        """2D np.float64 ndarray of column-proportion denominator for each cell."""
+        # --- Never add across subvariables. If rows dimension is an array, then the
+        # --- column_bases is the same as the counts.
+        if self._row_dimension_is_array:
+            return self.weighted_counts
+        # --- only selected counts contribute to the columns bases, which is summed
         # --- across the rows (axis 0).
-        return np.sum(self._weighted_counts[:, :, 0], axis=0)
+        return np.broadcast_to(
+            np.sum(self._weighted_counts[:, :, 0], axis=0), self.weighted_counts.shape
```

Comment:
why sum on the private _weighted_counts and the shape is from the public weighted_counts?